src/ug_sensors/ug_sensors/depth_sensor.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+"""
+    Originally written by:
+        Anthony Cieri penguinmillion@gmail.com
+
+    This creates a ros2 node that reads from the depth
+    sensor and publishes the readings to the sensor topic.
+
+    In its current state, this code can be used as a base for any
+    sensor node.
+
+    DONE: Create a ros2 node that publishes data to a topic.
+
+    TODO: Develop the sensor interface with David.
+
+    DONE: Get data from the sensor, requires appropriate libs
+            and hardware (raspberry pi and sensor).
+
+    TODO: Should a separate node be created for temperature?
+"""
+
+import rclpy  # For ros2
+from rclpy.node import Node  # To create the node
+from std_msgs.msg import (
+    Float32 as depth_sensor_interface,
+)  # TODO: Create topic interface with David
+
+# Sensor modules
+# This should be a module between the actual sensor driver and this script that
+# Translates whatever the sensor driver gives to what this script expects
+# TODO: What does this script expect? (Specify abstract sensor)
+import sensor.ms5837_sensor as sensor_driver
+
+
+"""
+    Init the depthSensorNode class and spin it.
+"""
+
+
+def main(args=None):
```

Comment:
Make sure you cleanup node. `ros_node.destroy_node()`
Cleaning up may not matter now but usually when you have a bunch of nodes cleaning it up is always good when/if the RaspberryPi does not have enough RAM.